modules/launcher/src/main/java/org/mule/module/launcher/log4j2/LoggerContextCache.java

```diff
@@ -64,6 +66,8 @@
     private static final long DEFAULT_DISPOSE_DELAY_IN_MILLIS = 15000;
 
     private final ArtifactAwareContextSelector artifactAwareContextSelector;
+    // Extra cache layer to avid some nasty implications for using guava cache at this point.
```

Comment:
nasty implications is not very descriptive. Since this is a very subtle bug in which a new developer could easily fall into again, I would add more documentation around this.... Maybe consider extracting this to a class with a generous javadoc